commit 962036db46a7e89de1725f6b902dae602c466d6b
Author: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date:   Tue Jan 29 16:24:41 2019 +0000

    staging: vc-sm-cma: Correct DMA configuration.
    
    Now that VCHIQ is setting up the DMA configuration as our
    parent device, don't try to configure it during probe.
    
    Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>

diff --git a/drivers/staging/vc04_services/vc-sm-cma/vc_sm.c b/drivers/staging/vc04_services/vc-sm-cma/vc_sm.c
index a8758ab8c87f..7b3ce1c205d6 100644
--- a/drivers/staging/vc04_services/vc-sm-cma/vc_sm.c
+++ b/drivers/staging/vc04_services/vc-sm-cma/vc_sm.c
@@ -703,9 +703,6 @@ static void vc_sm_connected_init(void)
 /* Driver loading. */
 static int bcm2835_vc_sm_cma_probe(struct platform_device *pdev)
 {
-	struct device *dev = &pdev->dev;
-	int err;
-
 	pr_info("%s: Videocore shared memory driver\n", __func__);
 
 	sm_state = kzalloc(sizeof(*sm_state), GFP_KERNEL);
@@ -714,13 +711,11 @@ static int bcm2835_vc_sm_cma_probe(struct platform_device *pdev)
 	sm_state->pdev = pdev;
 	mutex_init(&sm_state->map_lock);
 
-	dev->coherent_dma_mask = DMA_BIT_MASK(32);
-	dev->dma_mask = &dev->coherent_dma_mask;
-	err = of_dma_configure(dev, NULL, true);
-	if (err) {
-		dev_err(dev, "Unable to setup DMA: %d\n", err);
-		return err;
-	}
+	pdev->dev.dma_parms = devm_kzalloc(&pdev->dev,
+					   sizeof(*pdev->dev.dma_parms),
+					   GFP_KERNEL);
+	/* dma_set_max_seg_size checks if dma_parms is NULL. */
+	dma_set_max_seg_size(&pdev->dev, 0x3FFFFFFF);
 
 	vchiq_add_connected_callback(vc_sm_connected_init);
 	return 0;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1344_962036db46a7e89de1725f6b902dae602c466d6b.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1344_962036db46a7e89de1725f6b902dae602c466d6b.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
